R/indicator_native_range_year.R

```diff
@@ -0,0 +1,117 @@
+                                 x_lab = "year",
+                                 y_lab = "alien species",
+                                 relative = FALSE) {
+  
+  require(plotly)
+  require(data.table)
+  
+  type <- match.arg(type)
+  
+  if (is.null(jaartallen))
+    jaartallen <- sort(unique(data$first_observed))
+  
+  plotData <- data %>% 
+    mutate(native_continent = case_when(grepl(pattern = "Africa", native_range, ignore.case = TRUE) ~ "Africa",
+                                        grepl(pattern = "America", native_range, ignore.case = TRUE) ~ "America",
+                                        grepl(pattern = "Asia", native_range, ignore.case = TRUE) ~ "Asia",
+                                        grepl(pattern = "Australia", native_range, ignore.case = TRUE) ~ "Oceania",
+                                        grepl(pattern = "nesia", native_range, ignore.case = TRUE) ~ "Oceania",
+                                        grepl(pattern = "Europe", native_range, ignore.case = TRUE) ~ "Europe",
+                                        TRUE ~ as.character(NA)))
+    
+  plotData$locatie <- switch(type,
+                             native_range = plotData$native_range,
+                             native_continent = plotData$native_continent
+  )
+  
+  # Select data
+  plotData <- plotData[plotData$first_observed %in% jaartallen, c("first_observed", "locatie")]
+  plotData <- plotData[!is.na(plotData$first_observed) & !is.na(plotData$locatie), ]
+  
+  # Exclude unused provinces
+  plotData$locatie <- as.factor(plotData$locatie)    
+  plotData$locatie <- droplevels(plotData$locatie)
+  
+  # Summarize data per native_range and year
+  plotData$first_observed <- with(plotData, factor(first_observed, levels = 
+                                                     min(jaartallen):max(jaartallen)))
+  
+  summaryData <- melt(table(plotData), id.vars = "first_observed")
+  summaryData <- summaryData %>% 
+    group_by(first_observed) %>% 
+    mutate(total = sum(value),
+           perc = round((value/total)*100,2))
+  
+  # Summarize data per year
+  totalCount <- table(plotData$first_observed)
+  
+  
+  # For optimal displaying in the plot
+  summaryData$locatie <- as.factor(summaryData$locatie)
+  summaryData$locatie <- factor(summaryData$locatie, levels = rev(levels(summaryData$locatie)))
+  summaryData$first_observed <- as.factor(summaryData$first_observed)
+  
+  
+  
+  # Create plot
+  
+  if(relative == TRUE){
+    position <- "fill"
+    text <- paste0(summaryData$locatie, "<br>", summaryData$perc, "%")
+  }else{
+    position <- "stack"
+    text <- paste0(summaryData$locatie, "<br>", summaryData$value)
+  }
+  
+  pl <- ggplot(data = summaryData, aes(x = first_observed, y = value, fill = locatie, text = text)) +
+    geom_bar(position = position, stat = "identity") 
+  
+  if(relative == TRUE){
+    pl <- pl + scale_y_continuous(labels = scales::percent_format())
+  }
```

Comment:
As you use functions from package `ggplot2` you should add them to Roxygen documentation like here below:

```r
@importFrom ggplot2 ggplot, geom_bar, scale_y_continuous
```

Please check in all code if there are other gpglot2 functions to include in `@importFrom`.

Same for `scales::percent_format()`. Put `scales` pkg in `Imports` section of DESCRIPTION file and add:

```r
@importFrom scales percent_format
```
Then you don't need to specify `scales::` in the code :+1:

Reply:
done